source/examples/rapids-morpheus-pipeline/notebook.md

```diff
@@ -0,0 +1,164 @@
+for orchestration, scalability, and resiliency in a cloud-native environment.
+
+## Deployment Components
+
+The pipeline is deployed on Amazon EKS using several Kubernetes manifests:
+
+### Kafka Deployment (`k8s/kafka`)
+
+The Kafka cluster is deployed using the [Strimzi Operator](https://strimzi.io/), which simplifies Kafka deployment and
+management on Kubernetes. See instructions on section [Deploying on EKS](deploying-on-eks)
+
+The deployment configuration includes:
+
+- Kafka cluster setup `kafka-single-node.yaml`
+
+  This is a modification of the file [https://strimzi.io/examples/latest/kafka/kafka-single-node.yaml](https://strimzi.io/examples/latest/kafka/kafka-single-node.yaml) where we modify:
+
+  - Cluster name to `kafka-cluster`
+  - Modify the volume to use `type: ephemeral` and use `sizeLimit: 5Gi` (instead of `size: 100Gi` that corresponded to
+    `type: persistent-claim`)
+
+- Kafka topics setup
+- Kafka UI
+
+### Kafka Producer Deployment (`k8s/kafka-producer`)
+
+The Kafka producer is deployed as a separate pod using the `kafka-producer.yaml` manifest. It continuously generates
+and publishes network data to the Kafka topic. This producer script is containerized using a custom Docker image that can be built using the Dockerfile in the `scripts/producer-dockerfile` directory.
+
+- Uses `kafka-python` for message production.
+- Contains the producer script for generating network data.
+
+### Triton-Morpheus Deployment (`k8s/triton`)
+
+The inference server is deployed using the NVIDIA Morpheus- Triton Inference Server docker image
+`nvcr.io/nvidia/morpheus/morpheus-tritonserver-models:25.02`.
+
+### Morpheus Pipeline Deployment (`k8s/morpheus-pipeline`)
+
+The core processing pipeline is deployed as a separate pod that, uses an image that can be built using the Dockerfile in
+the `pipeline-dockerfile` directory.
+
+- Runs the Morpheus nightly 25.06 conda build
+- Contains all pipeline and stage scripts `scripts/pipeline-dockerfile/*.py`
+- Processes the streaming data through the various stages
+
+(deploying-on-eks)=
+
+## Deploying on EKS
+
+### Prerequisites
+
+You need to have the [`aws` CLI tool](https://aws.amazon.com/cli/) and [`eksctl` CLI tool](https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html) installed along with [`kubectl`](https://kubernetes.io/docs/tasks/tools/) for managing Kubernetes.
+
+### Launch GPU enabled EKS cluster
+
+We launch a GPU enabled EKS cluster with `eksctl`.
+
+```{note}
+1. You will need to create or import a public SSH key to be able to execute the following command.
+In your aws console under `EC2` in the side panel under Network & Security > Key Pairs, you can create a
+key pair or import (see "Actions" dropdown) one you've created locally.
+
+2. If you are not using your default AWS profile, add `--profile <your-profile>` to the following command.
+```
+
+```console
+$ eksctl create cluster morpheus-rapids \
+    --version 1.32 \
+    --nodes 3 \
```

Comment:
We only end up using 2 GPUs in this demo, one for Triton and one for Morpheus. Maybe we should default this to `2`.